py/objstr.c

```diff
@@ -743,11 +743,27 @@ STATIC mp_obj_t str_finder(size_t n_args, const mp_obj_t *args, int direction, b
     const mp_obj_type_t *self_type = mp_obj_get_type(args[0]);
     check_is_str_or_bytes(args[0]);
 
-    // check argument type
-    str_check_arg_type(self_type, args[1]);
-
     GET_STR_DATA_LEN(args[0], haystack, haystack_len);
-    GET_STR_DATA_LEN(args[1], needle, needle_len);
+
+    byte needle_data;
+    const byte *needle;
+    size_t needle_len;
+    if (mp_obj_is_int(args[1]) && (self_type == &mp_type_bytes
+                                   #if MICROPY_PY_BUILTINS_BYTEARRAY
+                                   || self_type == &mp_type_bytearray
+                                   #endif
+                                   )) {
+        // Allow {bytes/bytearray}.{find,index}(int).
+        needle_data = mp_obj_get_int(args[1]);
```

Comment:
Should this also check it's `0 <= value <= 255` ?

Reply:
Yes you're right. Adds back +16 bytes, but mostly it's "free" because the error message de-duplicates.